libretro/libretro.c

```diff
@@ -574,55 +617,12 @@ static void report_buttons (void)
 #endif
 }
 
-static void check_variables(void)
-{
-   bool reset_sfx = false;
-   struct retro_variable var;
-   var.key = "snes9x_next_overclock";
-   var.value = NULL;
-
-   if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
-   {
-      if (strcmp(var.value, "disabled") == 0)
-      {
-         Settings.SuperFXSpeedPerLine = 0.417 * 10.5e6;
-         reset_sfx = true;
-      }
-      else if (strcmp(var.value, "40MHz") == 0)
-      {
-         Settings.SuperFXSpeedPerLine = 0.417 * 40.5e6;
-         reset_sfx = true;
-      }
-      else if (strcmp(var.value, "60MHz") == 0)
-      {
-         Settings.SuperFXSpeedPerLine = 0.417 * 60.5e6;
-         reset_sfx = true;
-      }
-      else if (strcmp(var.value, "80MHz") == 0)
-      {
-         Settings.SuperFXSpeedPerLine = 0.417 * 80.5e6;
-         reset_sfx = true;
-      }
-      else if (strcmp(var.value, "100MHz") == 0)
-      {
-         Settings.SuperFXSpeedPerLine = 0.417 * 100.5e6;
-         reset_sfx = true;
-      }
-   }
-
-   if (reset_sfx)
-      S9xResetSuperFX();
-}
-
 void retro_run (void)
 {
    bool updated = false;
    poll_cb();
    report_buttons();
    S9xMainLoop();
-
-   if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE_UPDATE, &updated) && updated)
-      check_variables();
```

Comment:
Why,the over clock will still be applied on reset and I tested it and it
ran fine?
On Jun 6, 2016 12:26 PM, "Twinaphex" <notifications@github.com> wrote:

> In libretro/libretro.c
> <https://github.com/libretro/snes9x-next/pull/68#discussion_r65952521>:
>
> >  void retro_run (void)
> >  {
> >     bool updated = false;
> >     poll_cb();
> >     report_buttons();
> >     S9xMainLoop();
> > -
> > -   if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE_UPDATE, &updated) && updated)
> > -      check_variables();
>
> You shouldn't take this out. This needs to be called at least once per
> retro_run iteration.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/libretro/snes9x-next/pull/68/files/c31d18c6db0b043564332b047dddc27a1e1f6d8f..99069ef93f1f591aac052d321ae95b76779c5d45#r65952521>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABKgxafh29BorRbEB9QsesxDxNhzkKsdks5qJHR8gaJpZM4IhhiE>
> .
>